src/main/java/bookshop/dto/BookDto.java

```diff
@@ -1,14 +1,24 @@
 package bookshop.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookDto {
+    @NotNull
+    @Positive
     private Long id;
+    @NotBlank
     private String title;
```

Comment:
It's good to also validate a length of text fields, so it doesn't exceed DB column constraints.

Reply:
done